include/boost/beast/core/async_base.hpp

```diff
@@ -77,7 +77,7 @@ namespace beast {
     @code
 
     // Asynchronously read into a buffer until the buffer is full, or an error occurs
-    template<class AsyncReadStream, class ReadHandler>
+    template<class AsyncReadStream, BOOST_BEAST_ASYNC_TPARAM2 ReadHandler>
```

Comment:
This goes into the docs, I dont think we should be teaching `BOOST_BEAST_ASYNC_TPARAM2`